gpii/node_modules/flowManager/test/CaptureTests.js

```diff
@@ -0,0 +1,206 @@
+    jqUnit.assertEquals("There should be 2 solutions installed", 2, Object.keys(payload).length);
+    jqUnit.assertEquals("Check for Fake Mag 1", payload.fakemag1.name, "Fake Magnifier 1");
+    jqUnit.assertEquals("Check for Fake Mag 2", payload.fakemag2.name, "Fake Magnifier 2 - fully featured");
+};
+
+gpii.tests.flowManager.capture.adjustSolutions = function (solutionsRegistryDataSource) {
+    var testDataFile = __dirname + "/data/capture_fakemag_settings.json";
+
+    // The solutions block is a read only data structure
+    var adjustedSolutions = fluid.copy(solutionsRegistryDataSource.fullSolutionsRegistry);
+    adjustedSolutions.darwin.fakemag2.settingsHandlers.configuration.options.filename = testDataFile;
+    adjustedSolutions.darwin.fakemag1.settingsHandlers.configuration.options.filename = testDataFile;
+    adjustedSolutions.darwin.fakemag1.settingsHandlers.configuration1.options.filename = testDataFile;
+    solutionsRegistryDataSource.fullSolutionsRegistry = adjustedSolutions;
+};
+
+fluid.defaults("gpii.tests.flowManager.capture.tests", {
+    gradeNames: ["fluid.test.testEnvironment", "fluid.test.testCaseHolder"],
+    modules: [
+        {
+            name: "Simple system capture",
+            tests: [{
+                name: "Check for existing FakeMag Settings",
+                expect: 2,
+                sequence: [{
+                    funcName: "gpii.tests.flowManager.capture.adjustSolutions",
+                    args: ["{config}.server.flowManager.solutionsRegistryDataSource"]
+                }, {
+                    task: "{config}.server.flowManager.capture.getSystemSettingsCapture",
+                    args: [],
+                    resolve: "gpii.tests.flowManager.capture.checkForFakeMags",
+                    resolveArgs: ["{arguments}.0"]
+                }]
+            }]
+        },
+        {
+            name: "Simple installed solutions fetch",
+            tests: [{
+                name: "Testing get installed solutions",
+                expect: 3,
+                sequence: [{
+                    funcName: "gpii.tests.flowManager.capture.adjustSolutions",
+                    args: ["{config}.server.flowManager.solutionsRegistryDataSource"]
+                }, {
+                    task: "{config}.server.flowManager.capture.getInstalledSolutionsForCurrentDevice",
+                    args: [],
+                    resolve: "gpii.tests.flowManager.capture.checkForInstalledMags",
+                    resolveArgs: ["{arguments}.0"]
+                }]
+            }]
+        }
+    ],
+    components: {
+        "config": {
+            type: "gpii.flowManager.tests.capture.fakeData.config"
+        }
+    }
+});
+
+fluid.test.runTests([
+    "gpii.tests.flowManager.capture.tests"
+]);
+
+/*
+ *  Tests for formatting raw captures.
+ */
+gpii.tests.flowManager.capture.testFormatRawCapturedSettings = function () {
+    var allCorrectSettings = gpii.flowManager.formatRawCapturedSettings([
```

Comment:
A couple of thoughts:
1. It would be easier to read and maintain if all test cases are collected together. Just an example:

```
var testCases = {
    "allCorrectSettings": {
        message: "...",
        input: {...},
        expected: {...}
    },
    "withFetchError": {
        [same structure as above]
    },
    ...
};

fluid.each(testCases, function (oneTest) {
    var result = gpii.flowManager.formatRawCapturedSettings(oneTest.input);
    jqUnit.assert(oneTest.message, oneTest.expected, result);
});
```
2. Please also test edge cases such as, no settings are returned, all fetch errors etc.
3. Please use a slightly different setting sets/values especially when multiple settings are tested in one single test case so it doesn't miss the coding error when the same result is returned for all settings.

Reply:
Reworked these, and added more edge cases.